spec/02-integration/02-cmd/12-hybrid_spec.lua

```diff
@@ -0,0 +1,103 @@
+local helpers = require("spec.helpers")
+local x509 = require("resty.openssl.x509")
+local pl_file = require("pl.file")
+
+
+describe("kong hybrid", function()
+  lazy_setup(function()
+    helpers.get_db_utils(nil, {}) -- runs migrations
+    helpers.prepare_prefix()
+  end)
+
+  lazy_teardown(function()
+    helpers.clean_prefix()
+  end)
+
+  it("help", function()
+    local ok, stderr = helpers.kong_exec("hybrid --help")
+    assert.falsy(ok)
+    assert.not_equal("", stderr)
+  end)
+
+  describe("gen_cert", function()
+    it("gen_cert", function()
+      local cert = helpers.test_conf.prefix .. "/test1.crt"
+      local key = helpers.test_conf.prefix .. "/test1.key"
+
+      local ok, _, stdout = helpers.kong_exec("hybrid gen_cert " .. cert .. " " .. key)
+      assert.truthy(ok)
+      assert.matches("Successfully generated certificate/key pairs, they have been written to: ", stdout, nil, true)
+
+      assert.matches("-----BEGIN CERTIFICATE-----", pl_file.read(cert))
+      assert.matches("-----BEGIN PRIVATE KEY-----", pl_file.read(key))
+    end)
+
+    it("gen_cert sets correct permission on generated files", function()
+      local cert = helpers.test_conf.prefix .. "/test2.crt"
+      local key = helpers.test_conf.prefix .. "/test2.key"
+
+      local ok, _, stdout = helpers.kong_exec("hybrid gen_cert " .. cert .. " " .. key)
+      assert.truthy(ok)
+      assert.matches("Successfully generated certificate/key pairs, they have been written to: ", stdout, nil, true)
+
+      _, _, stdout = helpers.execute("ls -l " .. key)
+      assert.matches("-rw-------", stdout, nil, true)
+
+      _, _, stdout = helpers.execute("ls -l " .. cert)
+      assert.matches("-rw-r--r--", stdout, nil, true)
```

Comment:
these will fail if someone tries to run the testsuite with a tighter `umask` — but oh well, if somebody complains, we already know the answer ;)